### DISCUSSION TITLE
ansible-lint cannot find collection playbook

### DISCUSSION THREAD

**Discussion by jdratlif** [ansible/ansible-lint#2325] [General]
##### Summary

When I run ansible-lint, it reports an error on playbooks that import playbooks from collections.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
ansible --version
ansible [core 2.12.8]
  config file = None
  configured module search path = ['/home/jdratlif/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jdratlif/venvs/ansible_2.12.8/lib64/python3.8/site-packages/ansible
  ansible collection location = /home/jdratlif/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/jdratlif/.local/bin/ansible
  python version = 3.8.11 (default, Sep  1 2021, 12:33:46) [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
WARNING: PATH altered to include /home/jdratlif/venvs/ansible_2.12.8/bin
ansible-lint 6.5.0 using ansible 2.12.8
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### OS / ENVIRONMENT

CentOS 7

##### STEPS TO REPRODUCE

Create a collection with a playbook. Install the collection locally. Create playbook that uses import_playbook from that collection. Run ansible-lint --offline with playbook that imports playbook from collection. ansible-lint fails.

```console
- name: run base_qa playbook from collection
  ansible.builtin.import_playbook: globalnoc.base_qa.base_qa
```

##### Desired Behavior

ansible-lint to tell me the playbook is fine.

##### Actual Behavior

```console
ansible-lint --offline playbook.yml 
WARNING: PATH altered to include /home/jdratlif/venvs/ansible_2.12.8/bin
WARNING  Listing 1 violation(s) that are fatal
internal-error: Unexpected error code 1 from execution of: ansible-playbook --syntax-check -i localhost, playbook.yml
playbook.yml:1 ERROR! Unable to retrieve file contents
Could not find or access '/home/jdratlif/git/ansible/base-qa/globalnoc.base_qa.base_qa' on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option


You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - internal-error  # Unexpected internal error.

Finished with 1 failure(s), 0 warning(s) on 1 files.
```

```console
$ ansible-playbook --syntax-check -i localhost, playbook.yml 
[WARNING]: Could not match supplied host pattern, ignoring: qa_targets

playbook: playbook.yml
```

The collection playbook does exist on the server. I can run the playbook and it works fine.

**Comment by ssbarnea** [comment]
I tested manually and apparently that is not really a bug as there is no syntax error when the collection is present. I suspect you are running a different version of ansible from command line than the one that is found by ansible-lint.

Be sure that you create a `requirements.yml` file where you list the required collection, so ansible-lint will install it when you do not run with offline mode.

Also be sure you do not have extra config files or environment variables that might alter ansible configuration in such way that it make it look for collections in different locations.

FYI, the error that you are seeing comes directly from Ansible itself and not from the linter. The linter is just running ansible-playbook --syntax-check as a subprocess.

**Reply by jdratlif** [reply]
I think os_migrate does. For ansible openstack migrations. It at least has playbooks inside the collection. But I don't think they have other playbooks that import those playbooks.

https://galaxy.ansible.com/os_migrate/os_migrate

**Reply by ssbarnea** [reply]
You can also run extra verbose in order to see what linter does, add `-vv` to its call and watch for the logged info, it should give you a hint why ansible subprocess fails to find the collection.

**Reply by ssbarnea** [reply]
I personally installed `ansible.memsource` and tried to import `ansible.memsource.pull` and worked for me from the first attempt.
